firefly_client/firefly_client.py

```diff
@@ -653,16 +729,130 @@ def show_xyplot(self, tbl_id, standalone=False, **chart_params):
         chart_data_elements = [{'type': 'xycols', 'options': options, 'tblId': tbl_id}]
 
         cid = FireflyClient._gen_item_id('XYPlot')
-        if standalone:
-            group_id = 'default'
-        else:
-            group_id = tbl_id
+
+        if not group_id:
+            if standalone:
+                group_id = 'default'
+            else:
+                group_id = tbl_id
 
         payload = {'chartId': cid, 'chartType': 'scatter', 'groupId': group_id,
                    'chartDataElements': chart_data_elements}
 
         return self.dispatch_remote_action(self.channel, FireflyClient.ACTION_DICT['ShowXYPlot'], payload)
 
+    def show_histogram(self, tbl_id, group_id=None, **histogram_params):
+        """
+        Show a histogram
+
+        Parameters
+        ----------
+        tbl_id : `str`
+            A table ID of the data to be plotted.
+        group_id : `str`, optional
+            Group ID of the chart group where the histogram belongs to. If grid view is used, group id is the
+            cell id of the cell which contains the histogram.
+        \*\*histogram_params : optional keyword arguments
+            Parameters for histogram. The options are shown as below:
+
+            **col**: `str`
+                Column or expression to use for x values, can contain multiple column names,
+                ex. *log(col)* or *(col1-col2)/col3*.
+            **xOptions**: `str`
+                comma separated list of x axis options: flip,log.
+            **yOptions**: `str`
+                comma separated list of y axis options: flip,log.
+            **fixedBinSizeSelection**: {'binWidth', 'numBins'}
+                Fixed bin size selection.
```

Comment:
the function is updated like, 
- removing fixedBinSizeSelection from the API , it is set internally based on the setting of numBins and binWidth and sent with pushAction.
- checking falsePositiveRate, numBins, binWidth in order, if none of them are set, the default setting is algorithm='fixedSizeBins' , fixedBinSizeSelection='numBins', numBins=50.